src/transformState.js

```diff
@@ -5,7 +5,36 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const obj of actions) {
+    switch (true) {
+      case obj.type === 'addProperties':
+        addProperties(state, obj);
+        break;
+      case obj.type === 'removeProperties':
+        removeProperties(state, obj);
+        break;
+      default:
+        clear(state);
+    };
+  };
+};
+
+function addProperties(state, obj) {
+  Object.assign(state, obj.extraData);
+}
```

Comment:
if you're using small atomic functions it would be better to return the value at the end
```suggestion
function addProperties(state, obj) {
  Object.assign(state, obj.extraData);
  return state;
}
```

Reply:
Дякую, в цьому є сенс